backend/db_funcs/cipherGameFuncs.py

```diff
@@ -64,3 +64,29 @@ def getCipherGame(cipher_game_id: int):
     except:
         return {"result": "error"}
     return result
+
+
+def is_visible(cipher_game_id: int):
+    timestamp = datetime.now()
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute("SELECT * FROM cipher_game WHERE visible_from >=  %s AND cipher_game_id = %s;", (timestamp,cipher_game_id))
+                result = cur.fetchall()
+    except:
+        return False
+    return bool(result)
+
+
+def get_leaderboard(cipher_game_id: int):
+    if not is_visible(cipher_game_id):
+        return None
+    try:
+        with DB_conn.getConn(connection):
+            with DB_conn.getCursor(connection) as cur:
+                cur.execute("SELECT t.team_id, t.name, SUM(c.score) FROM team t JOIN attempt a ON a.team_id = t.team_id AND a.is_successful = TRUE JOIN cipher c ON c.cipher_id = a.cipher_id AND c.cipher_game_id = %s GROUP BY t.team_id, t.name, c.score;", cipher_game_id)
+                result = cur.fetchall()
```

Comment:
Chtěli bychom vracet konkrétně tohle:

Team ID, Jméno týmu, Skóre, Čas, Finální skóre (po odečtení hintů), Finální čas (po odečtení hintů).

Čas je čas teď (nebo čas vyřešení poslední šifry) minus čas otevření šifry, která je nastavená v databázi pro danou šifrovačku jako ta, se kterou se začne měřit čas. To asi nepůjde nějak hezky nacpat do jednoho SQL dotazu.

Finální skóre a finální čas bude jako klasické skóre a čas, jen odečteš skóre/přičteš penalizaci za každou šifru, kterou tým otevřel.